spec/interactors/send_password_reset_email_spec.rb

```diff
@@ -0,0 +1,23 @@
+require "rails_helper"
+
+RSpec.describe SendPasswordResetEmail do
+  describe ".call" do
+    let(:user) { create(:confirmed_user) }
+
+    subject(:send_email) do
+      described_class.call(user: user)
+    end
+
+    before do
+      CreatePasswordResetToken.call(user: user)
+      ActionMailer::Base.deliveries.clear
+    end
+
+    context "when successful" do
+      it "sends a password reset email" do
+        send_email
+        expect(ActionMailer::Base.deliveries.count).to eq(1)
```

Comment:
That's a fair point. I think this could go either way. On one side we rely on `ActionMailer` to handle this. This is not our code so it makes sense to just mock it and assume `deliver_now` does its thing. On the other side is good to have assurance that it actually is delivering. Although it's probably mocking deliveries inside the test environment. Since there's no other place we would test this functionality. 

hmm.... 

I think it's OK to leave the test to make sure the email was delivered. It goes against the single responsibility principle and test isolation but it's good to have that bit of extra assurance.